site/app/models/User.php

```diff
@@ -184,6 +187,18 @@ public function __construct(Core $core, $details = array()) {
         if (isset($details['grading_registration_sections'])) {
             $this->setGradingRegistrationSections($details['grading_registration_sections']);
         }
+
+        if (isset($details['time_zone'])) {
+            $this->time_zone = $details['time_zone'];
+        }
+    }
+
+    /**
+     * Set $this->time_zone
+     * @param string $time_zone Appropriate time zone string from DateUtils::getAvailableTimeZones()
+     */
+    public function setTimeZone(string $time_zone) {
+        $this->time_zone = $time_zone;
```

Comment:
Should be be implementing logic to verify that `$time_zone` is a valid string from `DateUtils::getAvailableTimeZones()` here?

Reply:
Yes.  I moved the validation step to this method.